src/metabase/models/collection.clj

```diff
@@ -936,19 +935,36 @@
 ;;; |                                              Personal Collections                                              |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
-(defn format-personal-collection-name
-  "Constructs the personal collection name from user name."
-  [first-name last-name]
-  (tru "{0} {1}''s Personal Collection" first-name last-name))
+(s/defn format-personal-collection-name :- su/NonBlankString
+  "Constructs the personal collection name from user name.
+  When displaying to users we'll tranlsate it to user's locale,
+  but to keeps things consistent in the database, we'll store the name in site's locale.
+
+  Practically, use `user-or-site` = `:site` when insert or update the name in database,
+  and `:user` when we need the name for displaying purposes"
+  [first-name last-name user-or-site]
+  (assert (#{:user :site} user-or-site))
```

Comment:
It's not super common to use `assert` directly for function parameter validation. If you want to have validation here you could use a [precondition](https://clojure.org/reference/special_forms#_fn_name_param_condition_map_expr_2) or spec or schema. (There are examples of all of these throughout the code so take your pick; we're not very consistent).

Reply:
I'll use pre-condition for this. Thanks for pointing that out.